client/webserver/webserver.go

```diff
@@ -108,20 +108,43 @@ func newMarketSyncer(ctx context.Context, cl *wsClient, feed *core.BookFeed) *de
 	return ssWaiter
 }
 
+// Run starts the marketSyncer listening for BookUpdates, which it relays to the
+// websocket client as notifications.
 func (m *marketSyncer) Run(ctx context.Context) {
 	defer m.feed.Close()
 out:
 	for {
 		select {
-		case update := <-m.feed.C:
+		case update, ok := <-m.feed.C:
+			if !ok {
+				// Should not happen, but don't spin furiously.
+				log.Warnf("marketSyncer stopping on feed closed")
+				return
+			}
+			if update.Action == core.FreshBookAction {
+				// For FreshBookAction, translate the *core.MarketOrderBook
+				// payload into a *marketResponse.
+				mob, ok := update.Payload.(*core.MarketOrderBook)
+				if !ok {
+					log.Errorf("FreshBookAction payload not a *MarketOrderBook")
+					continue
+				}
+				update.Payload = &marketResponse{
```

Comment:
Translation necessary as `marketResponse` is a `webserver` type, not a `core` type.  Does make some sense to have the feed consumers (e.g. `webserver` and `rpcserver`) use their own types as necessary I guess.